src/server/storage/storage.js

```diff
@@ -357,35 +361,53 @@ Storage.prototype.loadPaths = function (data, callback) {
 
     this.mongo.openProject(data.projectId)
         .then(function (dbProject) {
-            var loadedObjects = {};
-
-            //self.logger.error('paths:', data.paths);
+            var loadedObjects = {},
+                throttleDeferred = Q.defer(),
```

Comment:
An example:
The root is loaded and the meta nodes are all requested to be loaded:
/languageNode/metaNodePath1
/languageNode/metaNodePath2
/languageNode/metaNodePathN

Without the throttle the languageNode will be loaded N-times from the mongo database. With the throttle it is only loaded once. The loadObject-calls are already throttled for a one path (or what do you mean?).

Q doesn't have a throttle and I wouldn't like to use another library.